deps/deps.go

```diff
@@ -462,6 +467,53 @@ func GetConfig(ctx context.Context, layers []string, db *harmonydb.DB) (*config.
 	return curioConfig, nil
 }
 
+func updateBaseLayer(ctx context.Context, db *harmonydb.DB) error {
+	_, err := db.BeginTransaction(ctx, func(tx *harmonydb.Tx) (commit bool, err error) {
+		// Get existing base from DB
+		text := ""
+		err = tx.QueryRow(`SELECT config FROM harmony_config WHERE title=$1`, "base").Scan(&text)
+		if err != nil {
+			if strings.Contains(err.Error(), sql.ErrNoRows.Error()) {
+				return false, fmt.Errorf("missing layer 'base' ")
+			}
+			return false, fmt.Errorf("could not read layer 'base': %w", err)
+		}
+
+		cfg := config.DefaultCurioConfig()
+
+		// Apply existing base to updated default config
+		_, err = LoadConfigWithUpgrades(text, cfg)
+		if err != nil {
+			return false, fmt.Errorf("could not read base layer, bad toml %s: %w", text, err)
+		}
+
+		// Convert the update config to string
+		cb, err := config.ConfigUpdate(cfg, config.DefaultCurioConfig(), config.Commented(true), config.DefaultKeepUncommented(), config.NoEnv())
+		if err != nil {
+			return false, xerrors.Errorf("cannot update base config: %w", err)
+		}
```

Comment:
So here's a moderate issue - if a cluster has some mixed versions of nodes - new version introduced a new config, and it's in base - after an old node which is not aware of that field restarts, the new field would be removed from base. Probably fixable by working on toml.MetaData returned by LoadConfigWithUpgrades